src/pykeen/utils.py

```diff
@@ -1747,6 +1748,48 @@ def rate_limited(xs: Iterable[X], min_avg_time: float = 1.0) -> Iterable[X]:
         yield x
 
 
+class ExtraReprMixin:
+    """
+    A mixin for modules with hierarchical `extra_repr`.
+
+    It takes up the :meth:`torch.nn.Module.extra_repr` idea, and additionally provides a simple
+    composable way to generate the components of :meth:`extra_repr` via :meth:`iter_extra_repr`.
+
+    If combined with `torch.nn.Module`, make sure to put :class:`ExtraReprMixin` *behind*
+    :class:`torch.nn.Module` to prefer the latter's :func:`__repr__` implementation.
+    """
+
+    def iter_extra_repr(self) -> Iterable[str]:
+        """
+        Iterate over the components of the :meth:`extra_repr`.
+
+        This method is typically overridden. A common pattern would be
+
+        .. code-block:: python
+
+            def iter_extra_repr(self) -> Iterable[str]:
+                yield from super().iter_extra_repr()
+                yield "<key1>=<value1>"
+                yield "<key2>=<value2>"
+
+        :return:
+            an iterable over individual components of the :meth:`extra_repr`
+        """
+        return []
+
+    def extra_repr(self) -> str:
+        """
+        Generate the extra repr, cf. :meth`torch.nn.Module.extra_repr`.
+
+        :return:
+            the extra part of the :func:`repr`
+        """
+        return ", ".join(self.iter_extra_repr())
+
+    def __repr__(self) -> str:  # noqa: D105
+        return f"{self.__class__.__name__}({self.extra_repr()})"
```

Comment:
why doesn't this depend on the same repr as the nn.Module?

nn.Module also implements this:

```
def _get_name(self):
        return self.__class__.__name__
```

Reply:
Are you asking whether I want to re-implement the same logic as `torch.nn.Module.__repr__`? 

Because this model does not necessesarily derive from `torch.nn.Module`, e.g., when used for `KGInfo`. In that case, I implemented a simple `__repr__` which should match the previous output for, e.g., `KGInfo`. For those which derive from `torch.nn.Module`, we depend on it being earlier in the MRO, and thus having its `__repr__` be preferred.